diff -up texinfo-6.7.90/info/infomap.c.orig texinfo-6.7.90/info/infomap.c
--- texinfo-6.7.90/info/infomap.c.orig	2019-12-01 12:26:46.000000000 +0100
+++ texinfo-6.7.90/info/infomap.c	2021-02-24 12:56:06.865568572 +0100
@@ -589,6 +589,7 @@ fetch_user_maps (char *init_file)
   compile (inf, filename, &sup_info, &sup_ea);
 
   free (filename);
+  fclose (inf);
   return 1;
 }
 
diff -up texinfo-6.7.90/info/variables.c.orig texinfo-6.7.90/info/variables.c
--- texinfo-6.7.90/info/variables.c.orig	2021-02-24 13:00:21.056060523 +0100
+++ texinfo-6.7.90/info/variables.c	2021-02-24 13:36:27.089318922 +0100
@@ -359,6 +359,7 @@ read_variable_name (char *prompt, WINDOW
 {
   char *line;
   REFERENCE **variables;
+  VARIABLE_ALIST *alist;
 
   /* Get the completion array of variable names. */
   variables = make_variable_completions_array ();
@@ -382,7 +383,9 @@ read_variable_name (char *prompt, WINDOW
       return NULL;
     }
 
-  return variable_by_name (line);
+  alist = variable_by_name (line);
+  free (line);
+  return alist;
 }
 
 /* Make an array of REFERENCE which actually contains the names of the
diff -up texinfo-6.7.90/install-info/install-info.c.orig texinfo-6.7.90/install-info/install-info.c
--- texinfo-6.7.90/install-info/install-info.c.orig	2021-02-24 13:36:42.839472560 +0100
+++ texinfo-6.7.90/install-info/install-info.c	2021-02-24 13:41:36.219280631 +0100
@@ -1717,6 +1728,8 @@ reformat_new_entries (struct spec_entry
 
       format_entry (name, name_len, desc, desc_len, calign, align, 
                     maxwidth, &entry->text, &entry->text_len);
+      free (desc);
+      free (name);
     }
 }
 
